OSDOCS-10439 PID limit config support for ROSA with HCP (#75584)
openshift/openshift-docs: modules/setting-higher-pid-limit-on-machinepool.adoc

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa-configuring-pid-limits.adoc

:_mod-docs-content-type: PROCEDURE
[id="setting-higher-pid-limit-on-machine-pool_{context}"]
= Setting a higher process ID limit on a machine pool in a {product-title} cluster

You can set a higher `podPidsLimit` for machine pools in an existing {product-title} (ROSA) cluster by creating or editing a `KubeletConfig` object that changes the `--pod-pids-limit` parameter.

[IMPORTANT]
====

> Changing the `podPidsLimit` on an existing machine pool will trigger nodes in the machine pool to reboot one at a time. Make this change outside of peak usage hours for workloads in your machine pool and avoid upgrading or hibernating your cluster until all nodes have rebooted.

Reviewer: Do we want to avoid "will" here? https://spaces.redhat.com/display/CCSENABLE/Future+tense

```suggestion
Changing the `podPidsLimit` on an existing machine pool triggers nodes in the machine pool to reboot one at a time. Make this change outside of peak usage hours for workloads in your machine pool and avoid upgrading or hibernating your cluster until all nodes have rebooted.
```

Author: Agreed, thankyou!

====

.Prerequisites

* You have a {product-title} cluster.
* You have installed the ROSA CLI (`rosa`).
* You have logged in to your Red Hat account by using the ROSA CLI.

.Procedure

> . Create a new `KubeletConfig` object for your cluster that specifies a new `--pod-pids-limit`.

Reviewer: ```suggestion
. Create a new `KubeletConfig` object for your cluster that specifies a new `--pod-pids-limit`:
```

+
[source,terminal]
----
$ rosa create kubeletconfig -c <cluster_name> --name=<kubeletconfig_name>  --pod-pids-limit=<value>
----
+

> For example, the following command creates a `set-high-pids` KubeletConfig for the `my-cluster` cluster that sets a maximum of 16,384 PIDs per pod:

Reviewer 2: ```suggestion
For example, the following command creates a `set-high-pids` `KubeletConfig` object for the `my-cluster` cluster that sets a maximum of 16,384 PIDs per pod:
```

+
[source,terminal]
----
$ rosa create kubeletconfig -c my-cluster --name=set-high-pids --pod-pids-limit=16384
----

> . Associate the new `KubeletConfig` object with a new or existing machine pool.

Reviewer: Based on IBMSG "Procedure": https://www.ibm.com/docs/en/ibm-style?topic=punctuation-colons#introductory-text 

```suggestion
. Associate the new `KubeletConfig` object with a new or existing machine pool:
```

Author: Ignoring this one because the colon is attached to the conditional instructions, e.g. "For an existing machine pool:"

+
--
** For a new machine pool:
+
[source,terminal]
----
$ rosa create machinepool -c <cluster_name> --kubelet-configs=<kubeletconfig_name> --name <machinepool_name>
----
** For an existing machine pool:
+
[source,terminal]
----
$ rosa edit machinepool -c <cluster_name> --kubelet-configs=<kubeletconfig_name> --name <machinepool_name>
----
--
+

> For example, the following command associates the `set-high-pids` KubeletConfig with the `high-pid-pool` machine pool in the `my-cluster` cluster.

Reviewer 2: ```suggestion
For example, the following command associates the `set-high-pids` `KubeletConfig` object with the `high-pid-pool` machine pool in the `my-cluster` cluster.
```

> For example, the following command associates the `set-high-pids` `KubeletConfig` object with the `high-pid-pool` machine pool in the `my-cluster` cluster.

Reviewer: ```suggestion
For example, the following command associates the `set-high-pids` `KubeletConfig` object with the `high-pid-pool` machine pool in the `my-cluster` cluster:
```

+
[source,terminal]
----
$ rosa edit machinepool -c my-cluster --kubelet-configs=set-high-pids --name=high-pid-pool
----
+

> A rolling reboot of worker nodes is triggered when a new KubeletConfig is attached to an existing machine pool. You can check the progress of the rollout in the machine pool description:

Reviewer 2: ```suggestion
A rolling reboot of worker nodes is triggered when a new `KubeletConfig` object is attached to an existing machine pool. You can check the progress of the rollout in the machine pool description:
```

+
[source,terminal]
----
$ rosa describe machinepool --cluster <cluster_name> --name <machinepool_name>
----

.Verification

> * Confirm that the new setting is in place on nodes in the machine pool.

Reviewer: ```suggestion
* Confirm that the new setting is in place on nodes in the machine pool:
```
